### PR TITLE
[1494] Add location form

### PR DESCRIPTION
**Pull request by heidar**
### Context

We need to be able to add locations.

### Changes proposed in this pull request

Add form to add a new location.

### CODE REVIEW

**Comment by heidar** [comment]
[Allow users to create locations in mc-fe](https://trello.com/c/j4kzaBZn/1494-allow-users-to-create-locations-in-mc-fe)

File: app/views/sites/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      <li class="govuk-breadcrumbs__list-item">
+        <%= link_to "Locations", provider_sites_path(@provider[:provider_code]), class: "govuk-breadcrumbs__link" %>
+      </li>
+    </ol>
+  </nav>
+<% end %>
+
+<h1 class="govuk-heading-xl">Add a location</h1>
+
+<%= render 'errors' %>
```
Comment (dankmitchell) [inline app/views/sites/new.html.erb]: `<%= render 'errors'  if @errors.present? %>`

**Comment by tvararu** [comment]
@dankmitchell comments addressed

**Comment by tvararu** [comment]
Tested this locally and it works 😲

File: app/views/sites/index.html.erb [hunk 2]
```diff
@@ -49,7 +49,7 @@
         <%= render @sites %>
       </tbody>
     </table>
-    <%= add_location_link(current_user['info']['email'], @provider) %>
+    <%= link_to "Add a location", new_provider_site_path(@provider.provider_code), class: "govuk-button govuk-!-margin-bottom-2" %>
     <p class="govuk-body-s govuk-!-margin-bottom-6">You’ll be taken to a Google Form to fill in your location details.</p>
```
Comment (fofr) [inline app/views/sites/index.html.erb]: This text should be removed when the button is changed.

File: app/views/sites/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  <%= f.label :region_code, "Region of UK", class: "govuk-label" %>
+  <% if @errors[:region_code]&.any? then %>
+    <span id="region_code-error" class="govuk-error-message">
+      <span class="govuk-visually-hidden">Error:</span> <%= @errors[:region_code]&.first %>
+    </span>
+  <% end %>
+  <%= f.select :region_code, Site::REGIONS, {}, class: "govuk-select", aria: { describedby: @errors[:region_code] ? "region_code-error": '' } %>
+</div>
+
+<%= f.submit "Publish changes", class: "govuk-button", data: { qa: 'location__publish-changes' } %>
```
Comment (fofr) [inline app/views/sites/_form.html.erb]: When adding a location the button text should be: "Save"
When editing a location the button text should be "Save and publish changes"

(Technically the new location will be published, but as it's not attached to any course it's not visible anywhere and so publish can be confusing)

File: app/views/sites/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,34 @@
+<%= content_for :page_title, "#{@errors.any? ? "Error: " : ""}#{ @site_name_before_update }" %>
```
Comment (fofr) [inline app/views/sites/new.html.erb]: When you first go to the page the title comes out as:
`<title> - Publish teacher training courses - GOV.UK</title>`

It needs "Add a location" at the start. If a user enters content, hits save and there are errors, it should be "Error: Add a location"

Comment (tvararu) [reply app/views/sites/new.html.erb]: Done

File: app/views/sites/_form.html.erb (hunk 3 above)
Comment (tvararu) [reply app/views/sites/_form.html.erb]: Done

File: app/views/sites/index.html.erb (hunk 2 above)
Comment (tvararu) [reply app/views/sites/index.html.erb]: Done

**Pull request merged**